tests/foreman/ui/test_user.py

```diff
@@ -290,3 +292,45 @@ def test_positive_update_orgs(session):
         assert session.user.search(name)[0]['Username'] == name
         user = session.user.read(name)
         assert set(user['organizations']['resources']['assigned']) == set(org_names)
+
+
+@tier2
+def test_positive_create_product_with_limited_user_permission(
+    session, test_name, module_org, module_loc
+):
+    """A user with all permissions in Product and Repositories should be able to
+    create a new product
+
+    :id: 534a16f9-2d66-4fa1-aa5b-560f00eb4f67
+
+    :expectedresults: User successfully creates new product
+
+    :caseLevel: Component
+
+    :CaseImportance: High
+
+    :BZ: 1771937
+    """
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    product_name = gen_string('alpha')
+    product_label = gen_string('alpha')
+    product_description = gen_string('alpha')
+    role = entities.Role().create()
+    # Calling Products and Repositoy to get all the permissions in it
+    create_role_permissions(role, {'Katello::Product': PERMISSIONS['Katello::Product']})
+    entities.User(
+        default_organization=module_org,
+        organization=[module_org],
+        firstname='sample',
+        lastname='test',
+        role=[role],
+        login=username,
+        password=password,
+        mail='test@test.com',
+    ).create()
+    with Session(test_name, username, password) as newsession:
+        newsession.product.create(
+            {'name': product_name, 'label': product_label, 'description': product_description}
+        )
+        assert newsession.product.search(product_name)[0]['Name'] == product_name
```

Comment:
Citing the problem desc in the bz: "The product is created even with the error on step "Actions::Katello::Product::ReindexSubscriptions" and the task becomes on paused state and the product can not be deleted until cleaning up the task." 
Therefore I'd also attempt to delete the product.

Reply:
@JacobCallahan, cannot implement @pondrejk's suggestion because of bz: 1814311.  Airgun seems limited in it's functionality because it only deletes product once drilled in to the repo creation level.  Any suggestions on how to proceed, besides trying to modify airgun to delete at the product creation level?  This might go against the airgun design.

Reply:
@latran we can leave it be for now. can revisit once that bug has been resolved.